manim/animation/animation.py

```diff
@@ -112,21 +181,45 @@ def begin(self) -> None:
         self.interpolate(0)
 
     def finish(self) -> None:
+        # TODO: begin and finish should require a scene as parameter.
+        # That way Animation.clean_up_from_screen and Scene.add_mobjects_from_animations
+        # could be removed as they fulfill basically the same purpose.
+        """Finish the animation.
+
+        This method gets called when the animation is over.
+
+        """
         self.interpolate(1)
         if self.suspend_mobject_updating and self.mobject is not None:
             self.mobject.resume_updating()
 
     def clean_up_from_scene(self, scene: "Scene") -> None:
+        """Clean up the :class:`~.Scene` after finishing an animation.
+
+        This includes to :meth:`~.Scene.remove` the Animation's
+        :class:`~.Mobject` if the animation is a remover.
+
+        Parameters
+        ----------
+        scene
+            The scene the animation should be cleaned up from.
+        """
         if self.is_remover():
             scene.remove(self.mobject)
 
     def create_starting_mobject(self) -> Mobject:
         # Keep track of where the mobject starts
         return self.mobject.copy()
 
-    def get_all_mobjects(self) -> Tuple[Mobject, Mobject]:
-        """
+    def get_all_mobjects(self) -> Sequence[Mobject]:
```

Comment:
I think the previous typing gives more sense since it is always going to return a tuple where there are two Mobject.

Reply:
The reason I changed it, is that subclasses often return more than these two mobjects. Therefor I think Sequence[Mobject] is a better typehint, since it fits to the method in general and not only to the implementation in Animation. However, I don't have a strong opinion about this.

Reply:
Makes sense. But still, I could see https://github.com/ManimCommunity/manim/search?q=get_all_mobjects that each subclass returning a different thing, like some returns a Group and other a List. So, I think it is better to have `Tuple[Mobject, Mobject]` unless we change all the inherited subclass to return `Sequence[Mobject]`.

Reply:
I did the latter :)